ignite/metrics/running_average.py

```diff
@@ -139,32 +151,101 @@ def compute(self) -> Union[torch.Tensor, float]:
+                                                                        computed across batches in an epoch so it
+                                                                        is reset at the end of the epoch.
+                :class:`~.metrics.metric.RunningEpochWise`              Running average of the ``src`` metric or
+                                                                        ``engine.state.output`` is computed across
+                                                                        epochs. In the former case, ``src`` works
+                                                                        as if it was attached in a
+                                                                        :class:`~ignite.metrics.metric.EpochWise`
+                                                                        manner and its computed value is retrieved
+                                                                        at the end of the epoch. The latter case
+                                                                        doesn't make much sense for this usage as
+                                                                        the ``engine.state.output`` of the last
+                                                                        batch is retrieved then.
+                ======================================================= ===========================================
+
+        ``RunningAverage`` retrieves ``engine.state.output`` at ``usage.ITERATION_COMPLETED`` if the ``src`` is not
+        given and it's computed and updated using ``src``, by manually calling its ``compute`` method, or
+        ``engine.state.output`` at ``usage.COMPLETED`` event.
+        Also if ``src`` is given, it is updated at ``usage.ITERATION_COMPLETED``, but its reset event is determined by
+        ``usage`` type. If ``isinstance(usage, BatchWise)`` holds true, ``src`` is reset on ``BatchWise().STARTED``,
+        otherwise on ``EpochWise().STARTED`` if ``isinstance(usage, EpochWise)``.
+        """
+        usage = self._check_usage(usage)
+
+        if self.epoch_bound is not None:
+            warnings.warn(
+                "`epoch_bound` is deprecated and will be removed in the future. Consider using `usage` argument"
+                "instead. `epoch_bound=True` is equivalent with `usage=SingleEpochRunningBatchWise()` and "
+                "`epoch_bound=False` is equivalent with `usage=RunningBatchWise()`."
+            )
+            usage = SingleEpochRunningBatchWise() if self.epoch_bound else RunningBatchWise()
+
+        src_usage = EpochWise() if isinstance(usage, EpochWise) else BatchWise()
+        if isinstance(self.src, Metric) and not engine.has_event_handler(self.src.started, src_usage.STARTED):
+            engine.add_event_handler(src_usage.STARTED, self.src.started)
+            engine.add_event_handler(usage.ITERATION_COMPLETED, self.src.iteration_completed)
```

Comment:
@sadra-barikbin I'm not very sure to understand what you are doing here. Why do you need to define `src_usage`, can't 100% use `usage.STARTED` ? This looks a bit weird that we have to split like that `src_usage.STARTED` and `usage.ITERATION_COMPLETED`

Reply:
We could evade this only if reinstate `epoch_bound` and hardcode start event of the very `RunningAverage` to be `Events.STARTED if not epoch_bound else Events.EPOCH_STARTED`. Then we could designate `usage.STARTED` for `src` start event. If we do this we no longer need `Running......` usage classes and the `EpochWise` and `BatchWise` themselves serve as usage.

Reply:
All this logic seems a bit complicated to me. I have to dive deeper into the code to understand all the details.

How about this ?
```suggestion
        if isinstance(self.src, Metric):
            src_usage = EpochWise() if isinstance(usage, EpochWise) else BatchWise()
            if not engine.has_event_handler(self.src.started, src_usage.STARTED):
                engine.add_event_handler(src_usage.STARTED, self.src.started)
            engine.add_event_handler(usage.ITERATION_COMPLETED, self.src.iteration_completed)
```

Reply:
I changed the code. I think it is better now.